UploadBehavior.php

```diff
@@ -197,9 +197,8 @@ public function afterSave()
      */
     public function beforeDelete()
     {
-        $attribute = $this->attribute;
-        if ($this->unlinkOnDelete && $attribute) {
-            $this->delete($attribute);
+        if ($this->unlinkOnDelete) {
+            $this->deleteFolder($this->path);
```

Comment:
This is only true if the folder for each model is not unique.
However, this is not always correct.

E.g:

```php
class Document extends ActiveRecord
{
    function behaviors()
    {
        return [
            [
                'class' => UploadBehavior::class,
                'attribute' => 'file',
                'scenarios' => ['insert', 'update'],
                'path' => '@webroot/upload/docs',
                'url' => '@web/upload/docs',
            ],
        ];
    }
}
```

or all post attachments in the forum topic is stored in a shared folder

```php
class ForumPost extends ActiveRecord
{
    function behaviors()
    {
        return [
            [
                'class' => UploadBehavior::class,
                'attribute' => 'attachment',
                'scenarios' => ['insert', 'update'],
                'path' => '@webroot/upload/forum/{topic_id}',
                'url' => '@web/upload/forum/{topic_id}',
            ],
        ];
    }
}
```